### PR TITLE
Fix git ssh auth validation in gitrepository cr

### LINKED ISSUE #9383: Misleading error message for Git Functions when using ssh
**Description**

In new UI (not yet released) you can supply url to repo, which should be address to git repo with source code of function. Unfortunately in function-controller we do not explicitly say, that when we are using ssh (e.g. `git@github.com:kyma-project/kyma.git`) then we need to also provide a token, otherwise it bugs with following error 
image

**Expected result**
It doesn't allow me to create such a repository (preferably via webhook), or somehow informs me what is the cause of this error

**Actual result**
It gives me misleading info

**Steps to reproduce**
Provide any repo with function via ssh link with no credentials

**Troubleshooting**

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- add additional validation in case of using ssh in gitrepository.

**Related issue(s)**

fix #9383

### CODE REVIEW

File: components/function-controller/pkg/apis/serverless/v1alpha1/gitrepository_validation.go [hunk 1]
```diff
@@ -33,3 +38,27 @@ func (in *RepositoryAuth) validate(path string) *apis.FieldError {
...
+		err = apis.ErrGeneric(fmt.Sprintf("invalid value for git ssh, expected %s, current: %s",
+			RepositoryAuthSSHKey, in.Type), fmt.Sprintf("%s.type", path))
+	}
+
+	return err.Also(validateIfMissingFields(property{
+		name:  fmt.Sprintf("%s.secretName", path),
+		value: in.SecretName,
+	}))
+}
+func isRepoURLIsSSH(repoURL string) bool {
```
Comment (aerfio) [inline components/function-controller/pkg/apis/serverless/v1alpha1/gitrepository_validation.go:57]: Could you please write some more tests for that regexp? In first glance it looks kinda complicated

**Review by aerfio — CHANGES_REQUESTED** [review]

**Review by pPrecel — APPROVED** [review]
We should add more tests for the magic regexp in the `isRepoURLIsSSH` function :D But it looks great! Thanks for your contribution. 

Also, for me adding this validation to the webhook is a good option, but maybe someone has something against or better ideas? For example @m00g3n

File: components/function-controller/pkg/apis/serverless/v1alpha1/gitrepository_validation.go (hunk 1 above)
Comment (dbadura) [reply components/function-controller/pkg/apis/serverless/v1alpha1/gitrepository_validation.go:57]: Done

**Review by aerfio — APPROVED** [review]
LGTM now

**Pull request merged**